tests/unit/drivers/test_encoder_driver.py

```diff
@@ -22,35 +23,23 @@ def docs_to_encode(num_docs):
     return DocumentSet(docs)
 
 
-class MockEncoder(BaseEncoder):
-    def __init__(self,
-                 driver_batch_size: int,
-                 total_num_docs: int,
-                 *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.driver_batch_size = driver_batch_size
-        if self.driver_batch_size:
-            self.total_num_docs = total_num_docs
-            self.total_passes = int(self.total_num_docs / self.driver_batch_size)
-            self.id_pass = 0
+def get_encoder(batch_size):
+    class MockEncoder(BaseEncoder):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
 
-    @property
-    def is_last_batch(self):
-        return self.id_pass == self.total_passes
+        @batching(batch_size=batch_size)
+        def encode(self, data: Any, *args, **kwargs) -> Any:
+            if batch_size is not None and batch_size > 0:
+                assert len(data) <= batch_size
+            if batch_size == 5:
```

Comment:
this test is very weird, very dependant in the exact value of batch_size?

Reply:
The reason is that only for 1,2,5,10 you can guarantee for 10 documents, that each batch has a certain size. That's why I put this test only for the case of `batch_size` 5.